services/provider/client/client.go

```diff
@@ -188,13 +189,20 @@ func (cc *OCSProviderClient) GetStorageClassClaimConfig(ctx context.Context, con
 	return cc.Client.GetStorageClassClaimConfig(apiCtx, req)
 }
 
-func (cc *OCSProviderClient) ReportStatus(ctx context.Context, consumerUUID string) (*pb.ReportStatusResponse, error) {
+func NewStorageClientStatus() cs.StorageClientStatus {
+	return &pb.ReportStatusRequest{}
+}
+
+func (cc *OCSProviderClient) ReportStatus(ctx context.Context, consumerUUID string, status cs.StorageClientStatus) (*pb.ReportStatusResponse, error) {
 	if cc.Client == nil || cc.clientConn == nil {
 		return nil, fmt.Errorf("Provider client is closed")
 	}
-	req := &pb.ReportStatusRequest{
-		StorageConsumerUUID: consumerUUID,
+
+	req, ok := status.(*pb.ReportStatusRequest)
+	if !ok {
+		return nil, fmt.Errorf("Incorrect storage client status supplied")
```

Comment:
Do we want an error or a panic? 
I usually use panic where there is a coding error (non-runtime errors) to surface the coding mistake to the developer at an earlier stage and to make it more visible (similar to nil access panics)

Reply:
I do prefer panic, done.